src/parse.rs

```diff
@@ -100,30 +112,50 @@ impl LinkType {
     }
 }
 
+/// Markdown events that are generated in a preorder traversal of the document
+/// tree, with additional `End` events whenever all of an inner node's children
+/// have been visited.
 #[derive(Clone, Debug, PartialEq)]
 pub enum Event<'a> {
+    /// Start of a tagged element. Events that are yielded after this event
+    /// and before its corresponding `End` event are inside this element.
+    /// Start and end events are guaranteed to be balanced.
     Start(Tag<'a>),
+    /// End of a tagged element.
     End(Tag<'a>),
+    /// A text node.
     Text(CowStr<'a>),
+    /// An inline code node.
     Code(CowStr<'a>),
+    /// An HTML node.
     Html(CowStr<'a>),
-    InlineHtml(CowStr<'a>),
+    /// An inline HTML node.
+    InlineHtml(CowStr<'a>), // TODO: do we need a distinction between Html blocks/ Html/ InlineHtml?
+    /// A reference to a footnote with given label, which may or may not be defined
+    /// by an event with a `Tag::FootnoteDefinition` tag. Definitions and references to them may
+    /// occur in any order.
     FootnoteReference(CowStr<'a>),
+    /// A soft line break.
     SoftBreak,
+    /// A hard line break.
     HardBreak,
-    /// A task list marker, rendered as a checkbox in HTML. Contains a true when it is checked
+    /// A task list marker, rendered as a checkbox in HTML. Contains a true when it is checked.
     TaskListMarker(bool),
 }
 
+/// Table column text alignment.
 #[derive(Copy, Clone, Debug, PartialEq)]
 pub enum Alignment {
+    /// Default text alignment.
     None,
     Left,
     Center,
     Right,
 }
 
 bitflags! {
+    /// Option struct containing flags for enabling extra features
+    /// that are not part of the CommonMark spec.
     pub struct Options: u32 {
         const FIRST_PASS = 1 << 0;
```

Comment:
This should be removed next major release, as it's a source of confusion. But taking it out now would be a semver break. It might be worth adding a comment now saying, "this is unused and will be removed."

Reply:
Both of these are addressed in https://github.com/raphlinus/pulldown-cmark/pull/343 :-)

Will add that comment for now.